### PR TITLE
Use print() function in both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by cclauss**
**Note: DefectDojo is now on Python3 and Django 2.2. Please submit your pull requests to the 'python3_dev' branch as the 'dev' branch is only for bug fixes. Any PR's submitted to the 'dev' branch should also be converted to Python3 and submited to the 'python3_dev' branch. 

When submitting a pull request, please make sure you have completed the following checklist:

- [ ] Your code is flake8 compliant 
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ ] Model changes must include the necessary migrations in the dojo/dd_migrations folder.
- [ ] Add applicable tests to the unit tests.

### CODE REVIEW

File: dojo/management/commands/run_scan.py [hunk 1]
```diff
@@ -99,10 +100,10 @@ def runIPScan(host, service_dict):
             try:
                 nm = PortScannerAsync()
             except PortScannerError:
-                print('Nmap not found', sys.exc_info()[0])
+                print(('Nmap not found', sys.exc_info()[0]))
```
Comment (eamanu) [inline dojo/management/commands/run_scan.py]: I think that here is print a tuple

File: dojo/management/commands/run_scan.py [hunk 2]
```diff
@@ -99,10 +100,10 @@ def runIPScan(host, service_dict):
...
                 sys.exit(0)
             except:
-                print("Unexpected error:", sys.exc_info()[0])
+                print(("Unexpected error:", sys.exc_info()[0]))
```
Comment (eamanu) [inline dojo/management/commands/run_scan.py]: here the same

File: tests/zap.py [hunk 3]
```diff
@@ -100,11 +101,11 @@ class Main:
...
         summary.add_row(["Medium", medium])
         summary.add_row(["High", high])
-        print summary
+        print(summary)
 
         for url in sort_by_url:
-            print
-            print url
+            print()
+            print(url)
```
Comment (eamanu) [inline tests/zap.py]: what about ```print("\n" + url)``` and avoid the first  print?

File: dojo/management/commands/run_scan.py (hunk 1 above)
Comment (cclauss) [reply dojo/management/commands/run_scan.py]: You are correct.  Lines 204 and 212 contain legacy __print__ statements so we must assume that this file has not been ported to Python 3.  In Python 2, this line (without the import print_function) would print-a-tuple so now when we add the import print_function, we must ensure that it prints-a-tuple on both Python 2 and on Python 3.

Comment (eamanu) [reply dojo/management/commands/run_scan.py]: Oh yes, I was confused. Sorry for the noise. 

IMO this print should be just a str not a tuple. But well, that is a core decision

File: dojo/api.py [hunk 4]
```diff
@@ -1457,8 +1458,8 @@ def obj_create(self, bundle, **kwargs):
         try:
             t.full_clean()
         except ValidationError as e:
-            print "Error Validating Test Object"
-            print e
+            print("Error Validating Test Object")
```
Comment (eamanu) [inline dojo/api.py]: What about?

```suggestion
            print("Error Validating Test Object\n ".format(e))
```

File: dojo/engagement/views.py [hunk 5]
```diff
@@ -518,8 +519,8 @@ def import_scan_results(request, eid=None, pid=None):
 
             try:
                 for item in parser.items:
-                    print "item blowup"
-                    print item
+                    print("item blowup")
```
Comment (eamanu) [inline dojo/engagement/views.py]: what about?

```suggestion
                    print("item blowup\n{}".format(item))
```

File: dojo/api.py (hunk 4 above)
Comment (cclauss) [reply dojo/api.py]: That would halt the program.  The goal of this PR is not to optimize or improve the code.  We want to make safe changes to get it closer to Python 3 compatibility.

**Comment by devGregA** [comment]
Hi @cclauss, thank you for your PR! We have already converted Dojo to python 3.7 in the dev branch. The current master release is the second to last Python 2.7 release that will exist for Dojo. We have another legacy branch that will be the last 2.7 release once testing is completed on it. We don't accept PRs to the master brach only to dev per our PR template. Hope that all makes sense!

**Pull request closed without merging**